--- ../2.08fix-borscht/db_item.hsp	2022-01-28 15:34:53.304078500 -0800
+++ ../2.09-borscht/db_item.hsp	2022-01-28 18:58:56.744861700 -0800
@@ -1,25 +1,58 @@
 *db_item
 	if ( dbmode != DBMODE_FIND ) {
-		goto *label_3484
+		goto *label_3487
 	}
 	dbmax = 0
 	dbsum = 0
+	if ( 1 > objlv ) {
+		goto *label_0989
+	}
+	if ( fltselect != 1 ) {
+		goto *label_0989
+	}
+	if ( flttypemajor != (0 /*!!!@[FILTER_NOTHING @@@ FILTER_NOTHING]@!!! */) ) {
+		if ( flttypemajor != FILTER_ITEM_TOOL ) {
+			goto *label_0989
+		}
+	}
+	if ( flttypeminor != (0 /*!!!@[FILTER_NOTHING @@@ FILTER_NOTHING]@!!! */) ) {
+		if ( flttypeminor != FILTER_NONE ) {
+			goto *label_0989
+		}
+	}
+	if ( filtermax != 0 ) {
+		dbtmp = 1
+		repeat filtermax
+			if ( instr(filter_item(ITEM_ID_SOCKS2), 0, filtern(cnt)) == (-1) ) {
+				dbtmp = 0
+				break
+			}
+		loop
+		if ( dbtmp == 0 ) {
+			goto *label_0989
+		}
+	}
+	dbsum += 1000000 / (1000 + abs(objlv - 1) * 10) + 1
+	dblist(0, dbmax) = 1248, dbsum
+	dbmax++
+
+*label_0989
 	if ( 30 > objlv ) {
-		goto *label_0988
+		goto *label_0991
 	}
 	if ( fltselect != 0 ) {
-		goto *label_0988
+		goto *label_0991
 	}
 	if ( flttypemajor != (0 /*!!!@[FILTER_NOTHING @@@ FILTER_NOTHING]@!!! */) ) {
 		if ( flttypemajor != FILTER_ITEM_SPELLBOOK ) {
-			goto *label_0988
+			goto *label_0991
 		}
 	}
 	if ( flttypeminor != (0 /*!!!@[FILTER_NOTHING @@@ FILTER_NOTHING]@!!! */) ) {
 		if ( flttypeminor != FILTER_NONE ) {
-			goto *label_0988
+			goto *label_0991
 		}
 	}
 	if ( filtermax != 0 ) {
 		dbtmp = 1
 		repeat filtermax
@@ -41200,11 +41233,76 @@
 	else {
 		dbid = 25
 	}
 	return
 
-*label_3484
+*label_3487
+	if ( dbid == 1248 ) {
+		if ( dbmode == DBMODE_SET_STATS | dbmode == DBMODE_SET ) {
+			inv(INV_ITEM_VALUE, ci) = 300
+			inv(INV_ITEM_WEIGHT, ci) = 200
+		}
+		if ( dbmode == DBMODE_SET ) {
+			inv(INV_ITEM_ID, ci) = ITEM_ID_SOCKS2
+			inv(INV_ITEM_PIC, ci) = xy2pic(14, 36)
+			reftype = FILTER_ITEM_TOOL
+			reftypeminor = (0 /*!!!@[FILTER_NOTHING @@@ FILTER_NOTHING]@!!! */)
+			inv(INV_ITEM_NUM, ci) = 1
+			fixeditemenc = 0
+			inv(INV_ITEM_EFFECT, ci) = EFFECT_MEMORY_CONVERTER
+			inv(INV_ITEM_IDENTIFY_LEVEL, ci) = 0
+			return
+		}
+		if ( dbmode == DBMODE_REF ) {
+			ichargelevel = 0
+			infilterref = ""
+			reftype = FILTER_ITEM_TOOL
+			return
+		}
+		if ( dbmode == DBMODE_REF_SPEC ) {
+			if ( dbspec == 5 ) {
+				return 59000
+			}
+			if ( dbspec == 9 ) {
+				return 99999
+			}
+			if ( dbspec == 20 ) {
+				return 300
+			}
+			if ( dbspec == 19 ) {
+				return
+			}
+			if ( dbspec == 11 ) {
+				return 1
+			}
+			if ( dbspec == 18 ) {
+				return 0
+			}
+			if ( dbspec == 7 ) {
+				return 1
+			}
+			if ( dbspec == 8 ) {
+				return 1000
+			}
+			return 0
+		}
+		if ( dbmode == DBMODE_DESC ) {
+			if ( jp ) {
+				description(0) = "\t\t２枚１組の衣類。武器として使うことは想定されておらず防具にもならないが、こんなのでも収集しているマニアがいる。足を洗う習慣のない人々の靴下はかなり臭うので注意。ちなみに靴下を洗う際は、裏返して洗うと足に接していた部分の皮脂汚れが落ちやすくなる。\n#～パルミア秋冬物コレクション～\n"
+				description(1) = "\t\t「足を洗うなんてとんでもない」\n#～盗賊団の頭領の言葉～\n"
+				description(2) = "\t\t「靴下を洗うなんてとんでもないッ！！！！」\n#～脱ぎたて愛好家の言葉～\n"
+				description(3) = "\t\t臭う靴下。何度でも使用することができる。\n#～鑑定報告書：＜衣類＞カテゴリ～\n"
+			}
+			else {
+				description(0) = ""
+				description(1) = ""
+				description(3) = ""
+			}
+			return
+		}
+		return 0
+	}
 	if ( dbid == 1247 ) {
 		if ( dbmode == DBMODE_SET_STATS | dbmode == DBMODE_SET ) {
 			inv(INV_ITEM_VALUE, ci) = 4500
 			inv(INV_ITEM_WEIGHT, ci) = 380
 		}
@@ -41619,11 +41717,11 @@
 			gosub *readSpellbook
 			return -1
 		}
 		if ( dbmode == DBMODE_DESC ) {
 			if ( jp ) {
-				description(0) = "\t\t「混沌の渦」という呪文について学ぶことができる魔法書。夢見がちなあなたに。\n#～本の為の本・魔法書編～\n"
+				description(0) = "\t\t「夢幻の咆哮」という呪文について学ぶことができる魔法書。夢見がちなあなたに。\n#～本の為の本・魔法書編～\n"
 				description(1) = "<ランク7魔法>"
 				description(2) = ""
 				description(3) = "\t\t幻惑属性の範囲魔法呪文を唱える為に必要な本。読むことができる。\n#～鑑定報告書：＜書物＞カテゴリ～\n"
 			}
 			else {
@@ -41691,11 +41789,11 @@
 			gosub *readSpellbook
 			return -1
 		}
 		if ( dbmode == DBMODE_DESC ) {
 			if ( jp ) {
-				description(0) = "\t\t「轟音の波動」という呪文について学ぶことができる魔法書。サディストなあなたに。\n#～本の為の本・魔法書編～\n"
+				description(0) = "\t\t「苦悶の檻」という呪文について学ぶことができる魔法書。サディストなあなたに。\n#～本の為の本・魔法書編～\n"
 				description(1) = "<ランク7魔法>"
 				description(2) = ""
 				description(3) = "\t\t神経属性の範囲魔法呪文を唱える為に必要な本。読むことができる。\n#～鑑定報告書：＜書物＞カテゴリ～\n"
 			}
 			else {
@@ -42339,11 +42437,11 @@
 			gosub *readSpellbook
 			return -1
 		}
 		if ( dbmode == DBMODE_DESC ) {
 			if ( jp ) {
-				description(0) = "\t\t「幻惑の棘の吐息」という呪文について学ぶことができる魔法書。敵を幻覚で翻弄したいあなたに。\n#～本の為の本・魔法書編～\n"
+				description(0) = "\t\t「幻惑の棘」という呪文について学ぶことができる魔法書。敵を幻覚で翻弄したいあなたに。\n#～本の為の本・魔法書編～\n"
 				description(1) = "<ランク3魔法>"
 				description(2) = ""
 				description(3) = "\t\t幻惑属性の矢を唱える為に必要な本。読むことができる。\n#～鑑定報告書：＜書物＞カテゴリ～\n"
 			}
 			else {
@@ -123831,10 +123929,12 @@
 		return 0
 	}
 	return -999
 
 *db_setFilterItem
+	filter_item(ITEM_ID_SOCKS2) = ""
+	rffilter_item(ITEM_ID_SOCKS2) = ""
 	filter_item(ITEM_ID_SPELLBOOK_OF_THUNDER_VORTEX) = ""
 	rffilter_item(ITEM_ID_SPELLBOOK_OF_THUNDER_VORTEX) = ""
 	filter_item(ITEM_ID_SPELLBOOK_OF_ECLIPSE_JAIL) = ""
 	rffilter_item(ITEM_ID_SPELLBOOK_OF_ECLIPSE_JAIL) = ""
 	filter_item(ITEM_ID_SPELLBOOK_OF_NETHER_ROAR) = ""
@@ -126330,10 +126430,23 @@
 	filter_item(ITEM_ID_DUMMY) = ""
 	rffilter_item(ITEM_ID_DUMMY) = ""
 	return
 
 *db_setItem
+	ilight(ITEM_ID_SOCKS2) = LIGHT_TYPE_NONE
+	ialphanameref(ITEM_ID_SOCKS2) = ""
+	iorgweight(ITEM_ID_SOCKS2) = 200
+	if ( jp ) {
+		ioriginalnameref(ITEM_ID_SOCKS2) = "靴下"
+	}
+	else {
+		ioriginalnameref(ITEM_ID_SOCKS2) = "socks"
+		ioriginalnameref2(ITEM_ID_SOCKS2) = ""
+	}
+	ipicref(ITEM_ID_SOCKS2) = xy2pic(14, 36)
+	iknownnameref(ITEM_ID_SOCKS2) = ioriginalnameref(ITEM_ID_SOCKS2)
+	icolref(ITEM_ID_SOCKS2) = COLOR_RANDOM
 	ilight(ITEM_ID_SPELLBOOK_OF_THUNDER_VORTEX) = LIGHT_TYPE_NONE
 	ialphanameref(ITEM_ID_SPELLBOOK_OF_THUNDER_VORTEX) = ""
 	iorgweight(ITEM_ID_SPELLBOOK_OF_THUNDER_VORTEX) = 380
 	if ( jp ) {
 		ioriginalnameref(ITEM_ID_SPELLBOOK_OF_THUNDER_VORTEX) = "雷霆の渦の魔法書"
@@ -144211,11 +144324,11 @@
 		ioriginalnameref2(ITEM_ID_DUMMY) = ""
 	}
 	ipicref(ITEM_ID_DUMMY) = xy2pic(0, 1)
 	iknownnameref(ITEM_ID_DUMMY) = ioriginalnameref(ITEM_ID_DUMMY)
 	icolref(ITEM_ID_DUMMY) = COLOR_DEFAULT
-	maxitemid = 1250
+	maxitemid = 1251
 	return
 	goto *recipe_ref
 
 #deffunc calcweaponfix int calcweaponfix_itemid
 	pierce = 0
@@ -145533,11 +145646,12 @@
 		matval = SKILL_NORMAL_JEWELER, 1, 100
 		matneed = MATERIAL_PAPER, 2, MATERIAL_MEMORY_FRAGMENT, 1
 		return 1
 	}
 	return -1
-	goto *label_3493
+	goto *label_3496
+
 
 
 
 
 
