diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/action.hsp ../2.10-borscht/action.hsp
--- ../2.09-borscht/action.hsp	2022-01-28 18:58:54.064382500 -0800
+++ ../2.10-borscht/action.hsp	2022-01-30 13:05:31.159675500 -0800
@@ -4003,10 +4003,15 @@
 		msgdup++
 		txt lang(name(cc) + "は怖気づいた。", name(cc) + " " + is(cc) + " frightened.")
 		return
 	}
 	gdata(GDATA_FLAG_CURTAIN_CALL)++
+	if ( cdata(CDATA_GAUGE_COOLDOWN, cc) == 0 ) {
+		if ( cdata(CDATA_SP, cc) > (-50) ) {
+			cdata(CDATA_SP, cc)--
+		}
+	}
 	if ( cbit(CHARA_BIT_SANDBAG, tc) != TRUE ) {
 		if ( cdata(CDATA_GAUGE_COOLDOWN, cc) == 0 ) {
 			cdata(CDATA_GAUGE_COOLDOWN, cc) = 1
 			cdata(CDATA_POWER_GAUGE, cc) += 2
 			if ( cdata(CDATA_ID, cc) == CREATURE_ID_TALON_SOLDIER ) {
@@ -4065,10 +4070,18 @@
 				}
 				return
 			}
 		}
 	}
+	if ( cdata(CDATA_SP, cc) < 50 ) {
+		if ( rnd(2) == 0 ) {
+			if ( synccheck(cc, -1) ) {
+				txt lang("疲労し過ぎて失敗した！", "You are too exhausted!")
+			}
+			return
+		}
+	}
 	if ( attackrange == 1 ) {
 		animeid = attackskill
 		aniref = cw
 		gosub *anime
 	}
@@ -4636,20 +4649,43 @@
 		if ( cdata(CDATA_EXTRA_ATTACKS_TYPE, cc) == 0 ) {
 			if ( cdata(CDATA_EXTRA_ATTACKS, cc) != 0 ) {
 				if ( cdata(CDATA_EXIST, tc) == CHAR_STATE_ALIVE ) {
 					if ( orgdmg > 0 ) {
 						cw2 = -1
+						cw3 = -1
+						cw4 = -1
+						shootp = 0
 						repeat 30, 100
 							if ( cdata(cnt, cc) == 0 ) {
 								continue
 							}
 							if ( rnd(10) != 0 ) {
 								continue
 							}
 							cw2 = cdata(cnt, cc) - cdata(cnt, cc) / EXT_EQUIP_SLOTS * 10000 - 1
 							break
 						loop
+						repeat 30, 100
+							if ( cdata(cnt, cc) == 0 ) {
+								continue
+							}
+							if ( rnd(10) != 0 ) {
+								continue
+							}
+							cw3 = cdata(cnt, cc) - cdata(cnt, cc) / EXT_EQUIP_SLOTS * 10000 - 1
+							break
+						loop
+						repeat 30, 100
+							if ( cdata(cnt, cc) == 0 ) {
+								continue
+							}
+							if ( rnd(10) != 0 ) {
+								continue
+							}
+							cw4 = cdata(cnt, cc) - cdata(cnt, cc) / EXT_EQUIP_SLOTS * 10000 - 1
+							break
+						loop
 						if ( cw2 != (-1) ) {
 							repeat 15
 								if ( inv(40 + cnt * 2, cw2) == 0 ) {
 									break
 								}
@@ -4665,10 +4701,56 @@
 									cdata(CDATA_EXTRA_ATTACKS_TYPE, cc) = 2
 									shootp = inv(40 + cnt * 2 + 1, cw2)
 									break
 								}
 							loop
+							if ( shootp == 0 ) {
+								if ( cw3 != (-1) ) {
+									cw2 = cw3
+									repeat 15
+										if ( inv(40 + cnt * 2, cw2) == 0 ) {
+											break
+										}
+										enc = inv(40 + cnt * 2, cw2)
+										if ( dist(cdata(CDATA_X, cc), cdata(CDATA_Y, cc), cdata(CDATA_X, tc), cdata(CDATA_Y, tc)) <= 1 ) {
+											if ( enc == 66 ) {
+												cdata(CDATA_EXTRA_ATTACKS_TYPE, cc) = 1
+												shootp = inv(40 + cnt * 2 + 1, cw2)
+												break
+											}
+										}
+										if ( enc == 65 ) {
+											cdata(CDATA_EXTRA_ATTACKS_TYPE, cc) = 2
+											shootp = inv(40 + cnt * 2 + 1, cw2)
+											break
+										}
+									loop
+								}
+							}
+							if ( shootp == 0 ) {
+								if ( cw4 != (-1) ) {
+									cw2 = cw4
+									repeat 15
+										if ( inv(40 + cnt * 2, cw2) == 0 ) {
+											break
+										}
+										enc = inv(40 + cnt * 2, cw2)
+										if ( dist(cdata(CDATA_X, cc), cdata(CDATA_Y, cc), cdata(CDATA_X, tc), cdata(CDATA_Y, tc)) <= 1 ) {
+											if ( enc == 66 ) {
+												cdata(CDATA_EXTRA_ATTACKS_TYPE, cc) = 1
+												shootp = inv(40 + cnt * 2 + 1, cw2)
+												break
+											}
+										}
+										if ( enc == 65 ) {
+											cdata(CDATA_EXTRA_ATTACKS_TYPE, cc) = 2
+											shootp = inv(40 + cnt * 2 + 1, cw2)
+											break
+										}
+									loop
+								}
+							}
 							if ( cdata(CDATA_EXTRA_ATTACKS_TYPE, cc) != 0 ) {
 								if ( inv(INV_ITEM_QUALITY, cw2) >= FIX_QUALITY_MIRACLE ) {
 									if ( inv(INV_ITEM_QUALITY, cw2) == FIX_QUALITY_UNIQUE ) {
 										s(1) = lang("", "The ") + iknownnameref(inv(INV_ITEM_ID, cw2))
 									}
@@ -4794,10 +4876,17 @@
 								}
 								else {
 									s(1) = ioriginalnameref(inv(INV_ITEM_ID, cw2))
 								}
 							}
+							if ( shootp > 2000 ) {
+								shootp = 2000
+							}
+							if ( 0 ) {
+								cdata(CDATA_EXTRA_ATTACKS_TYPE, cc) = 1
+								shootp = 2000
+							}
 							if ( cdata(CDATA_EXTRA_ATTACKS_TYPE, cc) == 1 ) {
 								txtef COLOR_PINK
 								txt lang(name(cc) + "は" + s(1) + "で追撃した！", "Furthermore, " + name(cc) + " attacked with " + s(1) + "!")
 								if ( cdata(CDATA_PROT_PENALTY, tc) < 2 ) {
 									cdata(CDATA_PROT_PENALTY, tc) = 2
@@ -4806,11 +4895,11 @@
 								}
 								snd SOUNDLIST_PUNISH1
 								if ( cfg_attackanime ) {
 									animeload 39, tc
 								}
-								dmghp tc, (shootp / 40 * (orgdmg / 30) + rnd(sdata(SKILL_NORMAL_TACTICS, cc) * 10 + 1)) * 20 / limit(sdata(carmor(tc), tc) / 5 + 20, 20, 120) * 10, cc
+								dmghp tc, orgdmg / 30 * ((rnd(20) + shootp) / 50) + rnd(sdata(SKILL_NORMAL_TACTICS, cc) * 3 + 1), cc
 							}
 							if ( cdata(CDATA_EXTRA_ATTACKS_TYPE, cc) == 2 ) {
 								txtef COLOR_PINK
 								txt lang(name(cc) + "は" + s(1) + "による射撃を加えた！", "Furthermore, " + name(cc) + " fired " + s(1) + "!")
 								if ( cdata(CDATA_EVASION_PENALTY, tc) < 2 ) {
@@ -4820,11 +4909,11 @@
 								}
 								snd SOUNDLIST_PUNISH1
 								if ( cfg_attackanime ) {
 									animeload 39, tc
 								}
-								dmghp tc, (shootp / 50 * (orgdmg / 30) + rnd(sdata(SKILL_NORMAL_MARKSMAN, cc) * 5 + 1)) * 20 / limit(sdata(carmor(tc), tc) / 5 + 20, 20, 120) * 10, cc
+								dmghp tc, orgdmg / 30 * ((rnd(20) + shootp) / 50) + rnd(sdata(SKILL_NORMAL_MARKSMAN, cc) * 3 + 1), cc
 							}
 						}
 					}
 				}
 			}
@@ -5530,45 +5619,45 @@
 		}
 		if ( enc == 57 ) {
 			s = refchara(cdata(CDATA_ID, tc), DBSPEC_CHARA_FILTER, 1)
 			if ( instr(s, 0, "/dragon/") != (-1) ) {
 				gdata(GDATA_DMG_TYPE) = 1
-				dmghp tc, orgdmg / 30 * ((rnd(20) + inv(40 + cnt * 2 + 1, cw)) / 50) + 1, cc
+				dmghp tc, orgdmg / 30 * ((rnd(20) + limit(inv(40 + cnt * 2 + 1, cw), 1, 2000)) / 50) + 1, cc
 				subdmg = 0
 			}
 			continue
 		}
 		if ( enc == 61 ) {
 			s = refchara(cdata(CDATA_ID, tc), DBSPEC_CHARA_FILTER, 1)
 			if ( instr(s, 0, "/god/") != (-1) ) {
 				gdata(GDATA_DMG_TYPE) = 1
-				dmghp tc, orgdmg / 30 * ((rnd(20) + inv(40 + cnt * 2 + 1, cw)) / 50) + 1, cc
+				dmghp tc, orgdmg / 30 * ((rnd(20) + limit(inv(40 + cnt * 2 + 1, cw), 1, 2000)) / 50) + 1, cc
 				subdmg = 0
 			}
 			continue
 		}
 		if ( enc == 58 ) {
 			s = refchara(cdata(CDATA_ID, tc), DBSPEC_CHARA_FILTER, 1)
 			if ( instr(s, 0, "/undead/") != (-1) ) {
 				gdata(GDATA_DMG_TYPE) = 1
-				dmghp tc, orgdmg / 30 * ((rnd(20) + inv(40 + cnt * 2 + 1, cw)) / 50) + 1, cc
+				dmghp tc, orgdmg / 30 * ((rnd(20) + limit(inv(40 + cnt * 2 + 1, cw), 1, 2000)) / 50) + 1, cc
 				subdmg = 0
 			}
 			continue
 		}
 		if ( enc == 76 ) {
 			if ( cbit(CHARA_BIT_FLOAT, tc) == TRUE ) {
 				gdata(GDATA_DMG_TYPE) = 1
-				dmghp tc, orgdmg / 120 * ((rnd(10) + inv(40 + cnt * 2 + 1, cw)) / 50) + 1, cc
+				dmghp tc, orgdmg / 120 * ((rnd(10) + limit(inv(40 + cnt * 2 + 1, cw), 1, 2000)) / 50) + 1, cc
 				subdmg = 0
 			}
 			continue
 		}
 		if ( enc == 84 ) {
 			if ( cbit(CHARA_BIT_METAL, tc) == TRUE ) {
 				gdata(GDATA_DMG_TYPE) = 1
-				dmghp tc, orgdmg / 25 * ((rnd(100) + inv(40 + cnt * 2 + 1, cw)) / 50) + 100, cc
+				dmghp tc, orgdmg / 25 * ((rnd(100) + limit(inv(40 + cnt * 2 + 1, cw), 1, 2000)) / 50) + 100, cc
 				subdmg = 0
 			}
 			continue
 		}
 		i = enc / 10000
@@ -12927,11 +13016,14 @@
 	if ( _switch_val == 103 | _switch_sw ) {
 		_switch_sw = 0
 		inv(INV_ITEM_NUM, ci)--
 		cell_refresh inv(INV_ITEM_X, ci), inv(INV_ITEM_Y, ci)
 		snd SOUNDLIST_ATK_DARK
-		gdata(GDATA_FLAG_ABYSS_POWER) += 1
+		gdata(GDATA_FLAG_ABYSS_POWER) += 20
+		if ( gdata(GDATA_FLAG_ABYSS_POWER) > 100000 ) {
+			gdata(GDATA_FLAG_ABYSS_POWER) = 100000
+		}
 		txt lang("[深淵魔力: " + gdata(GDATA_FLAG_ABYSS_POWER) + "] ", "[Abyss power: " + gdata(GDATA_FLAG_ABYSS_POWER) + "] ")
 		goto *act_use_SWEND1
 		_switch_sw++
 	}
 	if ( _switch_val == 79 | _switch_sw ) {
diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/calculation.hsp ../2.10-borscht/calculation.hsp
--- ../2.09-borscht/calculation.hsp	2022-01-28 18:58:54.187496200 -0800
+++ ../2.10-borscht/calculation.hsp	2022-01-30 13:05:31.263771600 -0800
@@ -769,24 +769,24 @@
 	if ( locvar_calcinitgold_lootrich != (-1) ) {
 		return locvar_calcinitgold_lootrich * 500 + rnd(1000 + locvar_calcinitgold_lootrich * 1000)
 	}
 	return rnd(cdata(CDATA_LEVEL, calcinitgold_arg1) * 25 + 10) + 1
 
-*label_4035
-	goto *label_4037
+*label_4037
+	goto *label_4039
 
 #defcfunc calcspellpower int calcspellpower_skillid, int calcspellpower_charid
 	if ( calcspellpower_skillid >= (600 /*!!!@[MAX_SKILL_RANGE @@@ STARTING_SKILL_SPACT]@!!! */) ) {
 		if ( sdataref(SKILL_DATAREF_USE, calcspellpower_skillid) != SKILL_NONE ) {
-			return sdataref(SKILL_DATAREF_USE, calcspellpower_skillid) * 6 + 10
+			return sdataref(SKILL_DATAREF_USE, calcspellpower_skillid) * 10 + 50
 		}
 		return 100
 	}
 	if ( calcspellpower_charid == CHARA_PLAYER ) {
 		if ( gdata(GDATA_TEMPORARY_FLAGS) == 666 ) {
-			if ( sdata(calcspellpower_skillid, calcspellpower_charid) * 10 + 50 < sdata(SKILL_NORMAL_CASTING, calcspellpower_charid) * 6 + 10 ) {
-				return sdata(SKILL_NORMAL_CASTING, calcspellpower_charid) * 6 + 10
+			if ( sdata(calcspellpower_skillid, calcspellpower_charid) * 10 + 50 < sdata(SKILL_NORMAL_CASTING, calcspellpower_charid) * 10 + 50 ) {
+				return sdata(SKILL_NORMAL_CASTING, calcspellpower_charid) * 10 + 50
 			}
 		}
 	}
 	if ( calcspellpower_charid == CHARA_PLAYER ) {
 		return sdata(calcspellpower_skillid, calcspellpower_charid) * 10 + 50
@@ -794,14 +794,14 @@
 	if ( sdata(SKILL_NORMAL_CASTING, calcspellpower_charid) == 0 ) {
 		if ( calcspellpower_charid >= MAX_CHARA_FOLLOWER ) {
 			return cdata(CDATA_LEVEL, calcspellpower_charid) * 6 + 10
 		}
 	}
-	return sdata(SKILL_NORMAL_CASTING, calcspellpower_charid) * 7 + 10
+	return sdata(SKILL_NORMAL_CASTING, calcspellpower_charid) * 10 + 50
 
-*label_4037
-	goto *label_4039
+*label_4039
+	goto *label_4041
 
 #defcfunc calcspellfail int calcspellfail_skillid, int calcspellfail_charid
 	if ( calcspellfail_charid != CHARA_PLAYER ) {
 		if ( findbuff(calcspellfail_charid, BUFF_DISTRACTED) != (-1) ) {
 			if ( gdata(GDATA_RIDER) == calcspellfail_charid ) {
diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/chara_func.hsp ../2.10-borscht/chara_func.hsp
--- ../2.09-borscht/chara_func.hsp	2022-01-28 18:58:54.305605100 -0800
+++ ../2.10-borscht/chara_func.hsp	2022-01-30 13:05:31.385886800 -0800
@@ -2901,19 +2901,22 @@
 			}
 		}
 		return 1
 	}
 	if ( dmgcon_cond == CONDITION_INSANE ) {
-		locvar_dmgcon_p = dmgcon_power / 8 + 1
+		if ( cdata(CDATA_SANITY, dmgcon_charid) > 100 ) {
+			cdata(CDATA_SANITY, dmgcon_charid) = 100
+		}
+		locvar_dmgcon_p = dmgcon_power / 8 + cdata(CDATA_SANITY, dmgcon_charid)
 		if ( cdata(CDATA_CONDITION_DRUNK, dmgcon_charid) > 0 ) {
 			locvar_dmgcon_p /= 4
 		}
 		if ( cdata(CDATA_QUALITY, dmgcon_charid) > FIX_QUALITY_GREAT | cdata(CDATA_ID, dmgcon_charid) == CREATURE_ID_AILE_THE_ATTENDANT ) {
 			locvar_dmgcon_p /= 8
 		}
 		if ( findbuff(dmgcon_charid, BUFF_RETENTION) != (-1) ) {
-			locvar_dmgcon_p = 0
+			locvar_dmgcon_p /= 2
 		}
 		if ( locvar_dmgcon_p > 0 ) {
 			if ( cdata(CDATA_CONDITION_INSANE, dmgcon_charid) == 0 ) {
 				cdata(CDATA_CONDITION_INSANE, dmgcon_charid) = locvar_dmgcon_p
 				if ( synccheck(dmgcon_charid, -1) ) {
@@ -5596,13 +5599,12 @@
 			}
 		}
 	}
 	sutemi = 0
 	if ( dmghp_source >= (0 /*!!!@[CHARA_PLAYER @@@ DAMAGE_FROM_CHARA]@!!! */) ) {
-		if ( breakgauge == 0 ) {
+		if ( cdata(CDATA_GAUGE_COOLDOWN2, dmghp_charid) == 0 ) {
 			if ( cdata(CDATA_DRAIN_HP, dmghp_charid) < 100 ) {
-				breakgauge = 1
 				cdata(CDATA_DRAIN_HP, dmghp_charid) += 2
 				if ( locvar_dmghp_ele == 64 ) {
 					cdata(CDATA_DRAIN_HP, dmghp_charid) += 8
 				}
 				if ( cdata(CDATA_DRAIN_HP, dmghp_charid) >= 100 ) {
@@ -7345,13 +7347,10 @@
 	}
 	locvar_dmgsan_r = sdata(SKILL_RES_MIND, dmgsan_charid) / 50
 	if ( locvar_dmgsan_r <= 0 ) {
 		locvar_dmgsan_r = 1
 	}
-	if ( locvar_dmgsan_r > 10 ) {
-		return 0
-	}
 	locvar_dmgsan_dmg = dmgsan_arg2 / locvar_dmgsan_r
 	if ( dmgsan_charid < MAX_CHARA_FOLLOWER ) {
 		if ( trait(TRAIT_GOD_JURE) ) {
 			locvar_dmgsan_dmg = locvar_dmgsan_dmg - rnd(4) - 1
 		}
@@ -7363,10 +7362,13 @@
 		cdata(CDATA_SANITY, dmgsan_charid) += locvar_dmgsan_dmg
 	}
 	else {
 		locvar_dmgsan_dmg = 0
 	}
+	if ( cdata(CDATA_SANITY, dmgsan_charid) > 100 ) {
+		cdata(CDATA_SANITY, dmgsan_charid) = 100
+	}
 	if ( rnd(10) == 0 | rnd(locvar_dmgsan_dmg + 1) > 5 | rnd(cdata(CDATA_SANITY, dmgsan_charid) + 1) > 50 ) {
 		dmgcon dmgsan_charid, CONDITION_INSANE, 100
 	}
 	return 1
 
diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/chat.hsp ../2.10-borscht/chat.hsp
--- ../2.09-borscht/chat.hsp	2022-01-28 18:58:54.752017700 -0800
+++ ../2.10-borscht/chat.hsp	2022-01-30 13:05:31.878341000 -0800
@@ -37419,10 +37419,29 @@
 		}
 		goto *chat_end
 		goto *chat_unique_SWEND1
 		_switch_sw++
 	}
+	if ( _switch_val == 1056 | _switch_sw ) {
+		_switch_sw = 0
+		listmax = 0
+		buff = lang("なんですか、ジロジロと…わたしの顔に何かついてます？", "under development!")
+		tc = tc * (0 == 0) + (0 != 0) * 0
+		list(0, listmax) = 0
+		listn(0, listmax) = strbye
+		listmax++
+		chatesc = 1
+		gosub *chat_select
+		if ( scenemode ) {
+			if ( scene_cut == 1 ) {
+				goto *chat_end
+			}
+		}
+		goto *chat_end
+		goto *chat_unique_SWEND1
+		_switch_sw++
+	}
 	if ( _switch_val == 1024 | _switch_sw ) {
 		_switch_sw = 0
 		if ( gdata(GDATA_FLAG_MAIN) < 1000 ) {
 			listmax = 0
 			buff = lang("あはは。もし" + _onii(cdata(CDATA_SEX, CHARA_PLAYER)) + "さんが自分の物語を終わらせたら、アイメが楽しいお話をきかせてあげるねー。", "Sorry, this is untranslated sentence.")
diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/command.hsp ../2.10-borscht/command.hsp
--- ../2.09-borscht/command.hsp	2022-01-28 18:58:54.883139000 -0800
+++ ../2.10-borscht/command.hsp	2022-01-30 13:05:31.993449200 -0800
@@ -5714,11 +5714,11 @@
 		}
 	}
 	txt lang(name(tc) + "に何をする？ ", "What action do you want to perform to " + him(tc) + "? ")
 	kakudo2 = cdata(CDATA_MARK_ITEM_ADJUST_R, tc) * 15
 	if ( 0 ) {
-		txt lang("hxxxが" + cdata(CDATA_QUALITY, tc) + "でmhpが" + cdata(CDATA_MAX_HP, tc) + "。", "You cannot leave livestock on hanging.")
+		txt lang("hxxxが" + cdata(CDATA_QUALITY, tc) + "でmSpが" + cdata(CDATA_MAX_SP, tc) + "。", "You cannot leave livestock on hanging.")
 	}
 	if ( cdata(CDATA_PV_FIX, tc) == 100 & cdata(CDATA_DV_FIX, tc) == 100 ) {
 		if ( cdatan(CDATAN_RACE, tc) == "slime" ) {
 			cdata(CDATA_PV_FIX, tc) = 200
 		}
@@ -8971,12 +8971,15 @@
 				mes "Bad"
 				continue
 			}
 			mes "Hopeless"
 		loop
+		if ( cdata(CDATA_SANITY, cc) > 100 ) {
+			cdata(CDATA_SANITY, cc) = 100
+		}
 		if ( cc < MAX_CHARA_FOLLOWER ) {
-			s = "" + cdata(CDATA_HP, cc) + "/" + cdata(CDATA_MP, cc), "" + sdata(SKILL_ATTR_LIFE, cc) + "(" + sorg(SKILL_ATTR_LIFE, cc) + ")", "" + sdata(SKILL_ATTR_MANA, cc) + "(" + sorg(SKILL_ATTR_MANA, cc) + ")", "" + cdata(CDATA_SANITY, cc), "" + cdata(CDATA_SPEED, cc) + "(" + sorg(SKILL_ATTR_SPD, cc) + ")", "" + cdata(CDATA_FAME, cc), "" + cdata(CDATA_KARMA, CHARA_PLAYER), "" + cdata(CDATA_SHOP_INVEST, cc) + "" + lang("階相当", " level"), ""
+			s = "" + cdata(CDATA_HP, cc) + "/" + cdata(CDATA_MP, cc), "" + sdata(SKILL_ATTR_LIFE, cc) + "(" + sorg(SKILL_ATTR_LIFE, cc) + ")", "" + sdata(SKILL_ATTR_MANA, cc) + "(" + sorg(SKILL_ATTR_MANA, cc) + ")", "" + cdata(CDATA_SANITY, cc) + "" + lang("%", " %"), "" + cdata(CDATA_SPEED, cc) + "(" + sorg(SKILL_ATTR_SPD, cc) + ")", "" + cdata(CDATA_FAME, cc), "" + cdata(CDATA_KARMA, CHARA_PLAYER), "" + cdata(CDATA_SHOP_INVEST, cc) + "" + lang("階相当", " level"), ""
 		}
 		else {
 			if ( cdata(CDATA_ALLIED, cc) == 0 ) {
 				if ( cdata(CDATA_FAME, cc) == 0 ) {
 					cdata(CDATA_FAME, cc) = cdata(CDATA_LEVEL, cc) * 10 * cdata(CDATA_QUALITY, cc) * cdata(CDATA_QUALITY, cc) * cdata(CDATA_QUALITY, cc)
@@ -10692,12 +10695,20 @@
 	gcopy BUFFER_INF, 288 + 10 * 48, 48, 48, 48
 	pos wx + ww - 106, wy
 	gcopy BUFFER_INF, 960, 96, 96, 120
 	pos wx, wy + wh - 164
 	gcopy BUFFER_INF, 960, 216, 72, 144
-	dvr1 = cdata(CDATA_DV, cc) * cdata(CDATA_DV_FIX, cc) / 100 + cdata(CDATA_DV_FIX, cc) - 100
-	pvr1 = (cdata(CDATA_PV, cc) + cdata(CDATA_PV_FIX, cc) / 25) * cdata(CDATA_PV_FIX, cc) / 100
+	dvfix = cdata(CDATA_DV_FIX, cc)
+	pvfix = cdata(CDATA_PV_FIX, cc)
+	if ( cdata(CDATA_DV, cc) < 0 ) {
+		dvfix = 100
+	}
+	if ( cdata(CDATA_PV, cc) < 0 ) {
+		pvfix = 100
+	}
+	dvr1 = cdata(CDATA_DV, cc) * dvfix / 100 + dvfix - 100
+	pvr1 = (cdata(CDATA_PV, cc) + pvfix / 25) * pvfix / 100
 	if ( cdata(CDATA_DRAIN_HP, cc) >= 100 ) {
 		dvr1 = 0
 		pvr1 = 0
 	}
 	if ( showresist ) {
@@ -12000,12 +12011,20 @@
 		y = winposy(432) - 32
 		w = 200
 		h = 102
 		window x + 4, y + 4, w, h - h \ 8, 0, -1
 		window x, y, w, h - h \ 8, 0, 0
-		dvr1 = cdata(CDATA_DV, tc) * cdata(CDATA_DV_FIX, tc) / 100 + cdata(CDATA_DV_FIX, tc) - 100
-		pvr1 = (cdata(CDATA_PV, tc) + cdata(CDATA_PV_FIX, tc) / 25) * cdata(CDATA_PV_FIX, tc) / 100
+		dvfix = cdata(CDATA_DV_FIX, tc)
+		pvfix = cdata(CDATA_PV_FIX, tc)
+		if ( cdata(CDATA_DV, tc) < 0 ) {
+			dvfix = 100
+		}
+		if ( cdata(CDATA_PV, tc) < 0 ) {
+			pvfix = 100
+		}
+		dvr1 = cdata(CDATA_DV, tc) * dvfix / 100 + dvfix - 100
+		pvr1 = (cdata(CDATA_PV, tc) + pvfix / 25) * pvfix / 100
 		if ( cdata(CDATA_DRAIN_HP, tc) >= 100 ) {
 			dvr1 = 0
 			pvr1 = 0
 		}
 		font lang(cfg_font1, cfg_font2), 12 + en - en * 2, 0
@@ -12019,12 +12038,20 @@
 		y = winposy(432) - 32
 		w = 200
 		h = 102
 		window x + 4, y + 4, w, h - h \ 8, 0, -1
 		window x, y, w, h - h \ 8, 0, 0
-		dvr1 = cdata(CDATA_DV, tc) * cdata(CDATA_DV_FIX, tc) / 100 + cdata(CDATA_DV_FIX, tc) - 100
-		pvr1 = (cdata(CDATA_PV, tc) + cdata(CDATA_PV_FIX, tc) / 25) * cdata(CDATA_PV_FIX, tc) / 100
+		dvfix = cdata(CDATA_DV_FIX, tc)
+		pvfix = cdata(CDATA_PV_FIX, tc)
+		if ( cdata(CDATA_DV, tc) < 0 ) {
+			dvfix = 100
+		}
+		if ( cdata(CDATA_PV, tc) < 0 ) {
+			pvfix = 100
+		}
+		dvr1 = cdata(CDATA_DV, tc) * dvfix / 100 + dvfix - 100
+		pvr1 = (cdata(CDATA_PV, tc) + pvfix / 25) * pvfix / 100
 		if ( cdata(CDATA_DRAIN_HP, tc) >= 100 ) {
 			dvr1 = 0
 			pvr1 = 0
 		}
 		font lang(cfg_font1, cfg_font2), 12 + en - en * 2, 0
@@ -15036,11 +15063,11 @@
 	gosub *db_class
 	buff = ""
 	notesel buff
 	procid = (procid + 1) \ 4
 	proclist(procid) = "Dump:1" + " ct:" + ct
-	noteadd "Elona Version " + "6.09" + " キャラクター情報 " + gdata(GDATA_YEAR) + "年" + gdata(GDATA_MONTH) + "月" + gdata(GDATA_DAY) + "日 " + gdata(GDATA_HOUR) + "時" + gdata(GDATA_MIN) + "分  " + mdatan(0)
+	noteadd "Elona Version " + "6.10" + " キャラクター情報 " + gdata(GDATA_YEAR) + "年" + gdata(GDATA_MONTH) + "月" + gdata(GDATA_DAY) + "日 " + gdata(GDATA_HOUR) + "時" + gdata(GDATA_MIN) + "分  " + mdatan(0)
 	noteadd ""
 	noteadd "  " + fixtxt("" + cdatan(CDATAN_AKA, CHARA_PLAYER) + cdatan(CDATAN_NAME, CHARA_PLAYER), 34) + _sex(cdata(CDATA_SEX, CHARA_PLAYER)) + " " + calcage(0) + "歳" + "  " + cdata(CDATA_HEIGHT, CHARA_PLAYER) + "cm" + " " + cdata(CDATA_WEIGHT, CHARA_PLAYER) + "kg"
 	noteadd ""
 	noteadd fixtxt("種族       : " + racename, 30) + fixtxt("信仰      : " + godname(cdata(CDATA_GOD, CHARA_PLAYER)), 32)
 	noteadd fixtxt("職業       : " + classname, 30) + fixtxt("所属      : " + guildname(), 32)
diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/db_card.hsp ../2.10-borscht/db_card.hsp
--- ../2.09-borscht/db_card.hsp	2022-01-28 18:58:55.051294600 -0800
+++ ../2.10-borscht/db_card.hsp	2022-01-30 13:05:32.161963300 -0800
@@ -1,6 +1,19 @@
 *db_card
+	if ( dbid == 1056 ) {
+		cardreftype = 10
+		cardrefskill = lang("人間になりたくて神を信仰してきたツインテール。願いが漠然としすぎていて中途半端な人型になってしまった。当初は尻尾が頭についていることに違和感を抱いていたが、これはこれで便利だと気づいて今は満足している。", "")
+		cardrefdomain = 6
+		cardrefrare = 50
+		cardrefcost = 0
+		cardrefattack = 690
+		cardrefhp = 3
+		cardrefn = lang("ロングツインテールの『トーカ』", "<Talka> the long twintail")
+		cardrefpic = xy2pic(31, 28) + COLOR_DEFAULT * COLOR_TINT_MULT
+		cardrefrace = "1056 effect:" + skillname(cardrefattack) + " "
+		return 1
+	}
 	if ( dbid == 1055 ) {
 		cardreftype = 10
 		cardrefskill = lang("元々は人類を敵性体から守るための汎用人型防衛兵器。光の巨人に憧れた当時の人類がオーバーテクノロジーを結集して生み出した。それゆえにあんまり制御が効かず、結局闇堕ちして暴走している。", "")
 		cardrefdomain = 6
 		cardrefrare = 10
diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/help.hsp ../2.10-borscht/help.hsp
--- ../2.09-borscht/help.hsp	2022-01-28 18:58:56.932034900 -0800
+++ ../2.10-borscht/help.hsp	2022-01-30 13:05:34.230220400 -0800
@@ -5351,11 +5351,11 @@
 	}
 	if ( mode == MODELIST_QUICK_START ) {
 		repeat 9
 			gdata(STARTING_GDATA_RANK + cnt) = 5000
 		loop
-		gdata(GDATA_VERSION) = int(double("6.09") * 1000)
+		gdata(GDATA_VERSION) = int(double("6.10") * 1000)
 		gdata(STARTING_GDATA_SHORTCUT + 1) = 424
 		gdata(STARTING_GDATA_SHORTCUT + 2) = 300
 		gdata(STARTING_GDATA_SHORTCUT + 3) = 631
 		gdata(GDATA_COUNT_SHOP) = 1000
 		gdata(GDATA_COUNT_SHELTER) = 100
@@ -5958,11 +5958,11 @@
 		gdata(GDATA_AREA) = AREA_HOME
 		gdata(GDATA_HOME_LEVEL) = 0
 		gosub *setArea
 		gdata(GDATA_LEVEL) = 1
 		gdata(GDATA_LEVEL_START_ON) = 4
-		gdata(GDATA_VERSION) = int(double("6.09") * 1000)
+		gdata(GDATA_VERSION) = int(double("6.10") * 1000)
 		gdata(GDATA_WEATHER) = WEATHER_RAIN, 6
 		repeat 20
 			gdata(STARTING_GDATA_RANK + cnt) = 10000
 		loop
 		gosub *adv_init
@@ -7388,11 +7388,10 @@
 	}
 	return
 
 *turn_end
 	buffact = 0
-	breakgauge = 0
 	cc = ct
 	if ( cdata(CDATA_EXIST, cc) != CHAR_STATE_ALIVE ) {
 		goto *game_main
 	}
 	hear = 0
@@ -7783,11 +7782,11 @@
 		goto *pc_turn_init_WHILE1
 	}
 
 *pc_turn_init_WEND1
 	if ( dbg_titleinfo ) {
-		title "ElonaPlus 2.09" + " ver " + "6.09" + " " + exedir + " Seed: " + randseed + "  Loop:" + looplev + " SubLoop:" + sublev + " x:" + cdata(CDATA_X, CHARA_PLAYER) + " y:" + cdata(CDATA_Y, CHARA_PLAYER) + " ev:" + evnum + " q:" + gdata(GDATA_QUEST) + " tile:" + map(cdata(CDATA_X, CHARA_PLAYER), cdata(CDATA_Y, CHARA_PLAYER), MAP_TILE_ID) + " unpc:" + usernpcmax + "/" + gdata(GDATA_USERNPC_COUNT)
+		title "ElonaPlus 2.10" + " ver " + "6.10" + " " + exedir + " Seed: " + randseed + "  Loop:" + looplev + " SubLoop:" + sublev + " x:" + cdata(CDATA_X, CHARA_PLAYER) + " y:" + cdata(CDATA_Y, CHARA_PLAYER) + " ev:" + evnum + " q:" + gdata(GDATA_QUEST) + " tile:" + map(cdata(CDATA_X, CHARA_PLAYER), cdata(CDATA_Y, CHARA_PLAYER), MAP_TILE_ID) + " unpc:" + usernpcmax + "/" + gdata(GDATA_USERNPC_COUNT)
 	}
 	if ( trait(TRAIT_ETHER_ADDICTED_POTION) != 0 ) {
 		if ( rnd(5) == 0 ) {
 			inv_getheader 0
 			ci = rnd(invrange) + invhead
@@ -11754,10 +11753,17 @@
 		if ( gdata(GDATA_AREA) == AREA_ST_NORTH_BORDER ) {
 			flt
 			characreate -1, CREATURE_ID_ZISILION_THE_MINE_KING, 9, 10
 			cdata(CDATA_ROLE, rc) = ROLE_SPECIAL
 		}
+		if ( gdata(GDATA_AREA) == AREA_TRUCE_GROUND ) {
+			flt
+			characreate -1, CREATURE_ID_TALKA_THE_LONG_TWINTAIL, 16, 4
+			cdata(CDATA_ROLE, rc) = ROLE_SPECIAL
+			flt
+			itemcreate -1, ITEM_ID_AURTEHOM, 25, 16, 0
+		}
 		if ( gdata(GDATA_AREA) == AREA_PALMIA ) {
 			flt
 			characreate -1, CREATURE_ID_KURON_THE_BLACK_CLOTH, 51, 30
 			cdata(CDATA_ROLE, rc) = ROLE_SPECIAL
 		}
diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/init.hsp ../2.10-borscht/init.hsp
--- ../2.09-borscht/init.hsp	2022-01-28 18:58:57.144230800 -0800
+++ ../2.10-borscht/init.hsp	2022-01-30 13:05:34.374353300 -0800
@@ -975,18 +975,17 @@
 		if ( calcbuff_charid < CHARA_PLAYER ) {
 			if ( calcbuff_charid == CALCBUFF_DURATION ) {
 				return
 			}
 			else {
-				buffdesc = lang("PVを半減しさらに" + locvar_calcbuff_p2 + "減少", "Halves PV and decreases " + locvar_calcbuff_p + " ")
+				buffdesc = lang("PVを半減", "Halves PV")
 				return
 			}
 		}
 		if ( cdata(CDATA_PV, calcbuff_charid) > 0 ) {
 			cdata(CDATA_PV, calcbuff_charid) = cdata(CDATA_PV, calcbuff_charid) / 2
 		}
-		cdata(CDATA_PV, calcbuff_charid) -= locvar_calcbuff_p2
 		return
 	}
 	if ( calcbuff_buffid == BUFF_ELEMENT_SCAR ) {
 		dur = 20 + calcbuff_power / 5
 		if ( calcbuff_charid < CHARA_PLAYER ) {
@@ -1436,11 +1435,11 @@
 		if ( calcbuff_charid < CHARA_PLAYER ) {
 			if ( calcbuff_charid == CALCBUFF_DURATION ) {
 				return
 			}
 			else {
-				buffdesc = lang("テレポート妨害/速度上昇/耐狂気", "Prevent teleport/Increases speed/RES+ insane")
+				buffdesc = lang("テレポート妨害/速度上昇/狂気半減", "Prevent teleport/Increases speed/Halve insane")
 				return
 			}
 		}
 		sdata(SKILL_ATTR_SPD, calcbuff_charid) += sdata(SKILL_ATTR_SPD, calcbuff_charid) / 8
 		return
@@ -2095,77 +2094,77 @@
 	skillname(SKILL_SPELL_HYDRO_FANG) = lang("水の牙", "Hydro Fang")
 	skilldesc(SKILL_SPELL_HYDRO_FANG) = lang("PVDV属性の矢", "Target(PVDV)")
 	if ( 491 < 600 ) {
 		skillencdesc(SKILL_SPELL_HYDRO_FANG) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_CRYSTAL_SPEAR) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 10, 9000 + 6, 300
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_CRYSTAL_SPEAR) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 20, 9000 + 6, 300
 	skillname(SKILL_SPELL_CRYSTAL_SPEAR) = lang("マジックボルト", "Magic Bolt")
 	skilldesc(SKILL_SPELL_CRYSTAL_SPEAR) = lang("魔法属性のボルト", "Line(Magic)")
 	if ( 459 < 600 ) {
 		skillencdesc(SKILL_SPELL_CRYSTAL_SPEAR) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_ICE_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 14, 9000 + 6, 400
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_ICE_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 28, 9000 + 6, 400
 	skillname(SKILL_SPELL_ICE_BOLT) = lang("アイスボルト", "Ice Bolt")
 	skilldesc(SKILL_SPELL_ICE_BOLT) = lang("冷気属性のボルト", "Line(Cold)")
 	if ( 419 < 600 ) {
 		skillencdesc(SKILL_SPELL_ICE_BOLT) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_FIRE_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 14, 9000 + 6, 400
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_FIRE_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 28, 9000 + 6, 400
 	skillname(SKILL_SPELL_FIRE_BOLT) = lang("ファイアボルト", "Fire Bolt")
 	skilldesc(SKILL_SPELL_FIRE_BOLT) = lang("火炎属性のボルト", "Line(Fire)")
 	if ( 420 < 600 ) {
 		skillencdesc(SKILL_SPELL_FIRE_BOLT) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_LIGHTNING_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 14, 9000 + 6, 400
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_LIGHTNING_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 28, 9000 + 6, 400
 	skillname(SKILL_SPELL_LIGHTNING_BOLT) = lang("ライトニングボルト", "Lightning Bolt")
 	skilldesc(SKILL_SPELL_LIGHTNING_BOLT) = lang("雷属性のボルト", "Line(Lightning)")
 	if ( 421 < 600 ) {
 		skillencdesc(SKILL_SPELL_LIGHTNING_BOLT) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_DARKNESS_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 14, 9000 + 6, 500
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_DARKNESS_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 28, 9000 + 6, 500
 	skillname(SKILL_SPELL_DARKNESS_BOLT) = lang("ダークネスボルト", "Darkness Bolt")
 	skilldesc(SKILL_SPELL_DARKNESS_BOLT) = lang("暗黒属性のボルト", "Line(Darkness)")
 	if ( 422 < 600 ) {
 		skillencdesc(SKILL_SPELL_DARKNESS_BOLT) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_MIND_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 14, 9000 + 6, 500
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_MIND_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 28, 9000 + 6, 500
 	skillname(SKILL_SPELL_MIND_BOLT) = lang("マインドボルト", "Mind Bolt")
 	skilldesc(SKILL_SPELL_MIND_BOLT) = lang("幻惑属性のボルト", "Line(Mind)")
 	if ( 423 < 600 ) {
 		skillencdesc(SKILL_SPELL_MIND_BOLT) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_NETHER_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 14, 9000 + 6, 500
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_NETHER_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 28, 9000 + 6, 500
 	skillname(SKILL_SPELL_NETHER_BOLT) = lang("ネザーボルト", "Nether Bolt")
 	skilldesc(SKILL_SPELL_NETHER_BOLT) = lang("地獄属性のボルト", "Line(Nether)")
 	if ( 480 < 600 ) {
 		skillencdesc(SKILL_SPELL_NETHER_BOLT) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_POISON_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 14, 9000 + 6, 500
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_POISON_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 28, 9000 + 6, 500
 	skillname(SKILL_SPELL_POISON_BOLT) = lang("ポイズンボルト", "Poison Bolt")
 	skilldesc(SKILL_SPELL_POISON_BOLT) = lang("毒属性のボルト", "Line(Poison)")
 	if ( 481 < 600 ) {
 		skillencdesc(SKILL_SPELL_POISON_BOLT) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_SOUND_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 14, 9000 + 6, 500
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_SOUND_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 28, 9000 + 6, 500
 	skillname(SKILL_SPELL_SOUND_BOLT) = lang("サウンドボルト", "Sound Bolt")
 	skilldesc(SKILL_SPELL_SOUND_BOLT) = lang("音属性のボルト", "Line(Sound)")
 	if ( 482 < 600 ) {
 		skillencdesc(SKILL_SPELL_SOUND_BOLT) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_CHAOS_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 14, 9000 + 6, 500
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_CHAOS_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 28, 9000 + 6, 500
 	skillname(SKILL_SPELL_CHAOS_BOLT) = lang("カオスボルト", "Chaos Bolt")
 	skilldesc(SKILL_SPELL_CHAOS_BOLT) = lang("混沌属性のボルト", "Line(Chaos)")
 	if ( 483 < 600 ) {
 		skillencdesc(SKILL_SPELL_CHAOS_BOLT) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_NERVE_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 14, 9000 + 6, 500
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_NERVE_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 28, 9000 + 6, 500
 	skillname(SKILL_SPELL_NERVE_BOLT) = lang("ナーブボルト", "Nerve Bolt")
 	skilldesc(SKILL_SPELL_NERVE_BOLT) = lang("神経属性のボルト", "Line(Nerve)")
 	if ( 484 < 600 ) {
 		skillencdesc(SKILL_SPELL_NERVE_BOLT) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_WATER_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 10, 9000 + 6, 600
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_WATER_BOLT) = SKILL_ATTR_MAG, SKILL_TYPE_BOLT, 20, 9000 + 6, 600
 	skillname(SKILL_SPELL_WATER_BOLT) = lang("ハイドロボルト", "Hydro Bolt")
 	skilldesc(SKILL_SPELL_WATER_BOLT) = lang("PVDV属性のボルト", "Line(PVDV)")
 	if ( 475 < 600 ) {
 		skillencdesc(SKILL_SPELL_WATER_BOLT) = lang("", "")
 	}
@@ -2173,77 +2172,77 @@
 	skillname(SKILL_SPACT_SUICIDE_ATTACK) = lang("自爆", "Suicide Attack")
 	skilldesc(SKILL_SPACT_SUICIDE_ATTACK) = lang("PVDV属性の範囲攻撃", "Suicide(PVDV)")
 	if ( 644 < 600 ) {
 		skillencdesc(SKILL_SPACT_SUICIDE_ATTACK) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_MAGIC_STORM) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 15, 3000 + 2, 500
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_MAGIC_STORM) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 40, 3000 + 2, 500
 	skillname(SKILL_SPELL_MAGIC_STORM) = lang("魔力の嵐", "Magic Storm")
 	skilldesc(SKILL_SPELL_MAGIC_STORM) = lang("魔法属性の範囲攻撃", "Surround(Magic)")
 	if ( 460 < 600 ) {
 		skillencdesc(SKILL_SPELL_MAGIC_STORM) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_ICE_BALL) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 21, 3000 + 2, 600
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_ICE_BALL) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 56, 3000 + 2, 600
 	skillname(SKILL_SPELL_ICE_BALL) = lang("氷結の波動", "Freezing Wave")
 	skilldesc(SKILL_SPELL_ICE_BALL) = lang("冷気属性の範囲攻撃", "Surround(Cold)")
 	if ( 431 < 600 ) {
 		skillencdesc(SKILL_SPELL_ICE_BALL) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_FIRE_BALL) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 21, 3000 + 2, 600
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_FIRE_BALL) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 56, 3000 + 2, 600
 	skillname(SKILL_SPELL_FIRE_BALL) = lang("灼熱の嵐", "Burning Storm")
 	skilldesc(SKILL_SPELL_FIRE_BALL) = lang("火炎属性の範囲攻撃", "Surround(Fire)")
 	if ( 432 < 600 ) {
 		skillencdesc(SKILL_SPELL_FIRE_BALL) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_THUNDER_VORTEX) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 21, 3000 + 2, 600
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_THUNDER_VORTEX) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 56, 3000 + 2, 600
 	skillname(SKILL_SPELL_THUNDER_VORTEX) = lang("雷霆の渦", "Thunder Vortex")
 	skilldesc(SKILL_SPELL_THUNDER_VORTEX) = lang("電撃属性の範囲攻撃", "Surround(Lightning)")
 	if ( 470 < 600 ) {
 		skillencdesc(SKILL_SPELL_THUNDER_VORTEX) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_CHAOS_BALL) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 21, 3000 + 2, 700
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_CHAOS_BALL) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 56, 3000 + 2, 700
 	skillname(SKILL_SPELL_CHAOS_BALL) = lang("混沌の渦", "Chaos Vortex")
 	skilldesc(SKILL_SPELL_CHAOS_BALL) = lang("混沌属性の範囲攻撃", "Surround(Chaos)")
 	if ( 433 < 600 ) {
 		skillencdesc(SKILL_SPELL_CHAOS_BALL) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_RAGING_ROAR) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 21, 3000 + 2, 700
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_RAGING_ROAR) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 56, 3000 + 2, 700
 	skillname(SKILL_SPELL_RAGING_ROAR) = lang("轟音の波動", "Howling Wave")
 	skilldesc(SKILL_SPELL_RAGING_ROAR) = lang("轟音属性の範囲攻撃", "Surround(Sound)")
 	if ( 434 < 600 ) {
 		skillencdesc(SKILL_SPELL_RAGING_ROAR) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_ECLIPSE) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 21, 3000 + 2, 700
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_ECLIPSE) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 56, 3000 + 2, 700
 	skillname(SKILL_SPELL_ECLIPSE) = lang("月蝕の檻", "Eclipse Jail")
 	skilldesc(SKILL_SPELL_ECLIPSE) = lang("暗黒属性の範囲攻撃", "Surround(Darkness)")
 	if ( 469 < 600 ) {
 		skillencdesc(SKILL_SPELL_ECLIPSE) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_NETHER_WAVE) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 21, 3000 + 2, 700
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_NETHER_WAVE) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 56, 3000 + 2, 700
 	skillname(SKILL_SPELL_NETHER_WAVE) = lang("冥王の咆哮", "Nether Roar")
 	skilldesc(SKILL_SPELL_NETHER_WAVE) = lang("地獄属性の範囲攻撃", "Surround(Nether)")
 	if ( 471 < 600 ) {
 		skillencdesc(SKILL_SPELL_NETHER_WAVE) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_POISON_STORM) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 21, 3000 + 2, 700
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_POISON_STORM) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 56, 3000 + 2, 700
 	skillname(SKILL_SPELL_POISON_STORM) = lang("猛毒の嵐", "Poison Storm")
 	skilldesc(SKILL_SPELL_POISON_STORM) = lang("毒属性の範囲攻撃", "Surround(Poison)")
 	if ( 473 < 600 ) {
 		skillencdesc(SKILL_SPELL_POISON_STORM) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_ILLUSION_ROAR) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 21, 3000 + 2, 700
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_ILLUSION_ROAR) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 56, 3000 + 2, 700
 	skillname(SKILL_SPELL_ILLUSION_ROAR) = lang("夢幻の咆哮", "Illusion Roar")
 	skilldesc(SKILL_SPELL_ILLUSION_ROAR) = lang("幻惑属性の範囲攻撃", "Surround(Mind)")
 	if ( 478 < 600 ) {
 		skillencdesc(SKILL_SPELL_ILLUSION_ROAR) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_ANGUISH_JAIL) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 21, 3000 + 2, 700
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_ANGUISH_JAIL) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 56, 3000 + 2, 700
 	skillname(SKILL_SPELL_ANGUISH_JAIL) = lang("苦悶の檻", "Anguish Jail")
 	skilldesc(SKILL_SPELL_ANGUISH_JAIL) = lang("神経属性の範囲攻撃", "Surround(Nerve)")
 	if ( 479 < 600 ) {
 		skillencdesc(SKILL_SPELL_ANGUISH_JAIL) = lang("", "")
 	}
-	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_BUBBLE_STORM) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 15, 3000 + 2, 800
+	sdataref(SKILL_DATAREF_USE, SKILL_SPELL_BUBBLE_STORM) = SKILL_ATTR_MAG, SKILL_TYPE_BALL, 40, 3000 + 2, 800
 	skillname(SKILL_SPELL_BUBBLE_STORM) = lang("泡沫の嵐", "Bubble Storm")
 	skilldesc(SKILL_SPELL_BUBBLE_STORM) = lang("PVDV属性の範囲攻撃", "Surround(PVDV)")
 	if ( 476 < 600 ) {
 		skillencdesc(SKILL_SPELL_BUBBLE_STORM) = lang("", "")
 	}
@@ -19728,11 +19727,11 @@
 		goto *exit_game
 	}
 	else {
 		errorhack++
 	}
-	buf = "HspError " + wparam + " WinError " + GetLastError() + " Ver " + int(double("6.09") * 1000) + " Mode " + mode + "\n"
+	buf = "HspError " + wparam + " WinError " + GetLastError() + " Ver " + int(double("6.10") * 1000) + " Mode " + mode + "\n"
 	buf += "efId " + efid + " efP " + efp + " Area " + gdata(GDATA_AREA) + "/" + gdata(GDATA_LEVEL) + "\n"
 	buf += "ci "
 	if ( ci < 200 ) {
 		buf += "0"
 		if ( ci >= 0 ) {
diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/item.hsp ../2.10-borscht/item.hsp
--- ../2.09-borscht/item.hsp	2022-01-28 18:58:57.226306800 -0800
+++ ../2.10-borscht/item.hsp	2022-01-30 13:05:34.460432800 -0800
@@ -3509,15 +3509,11 @@
 			modweight cc, rnd(3) + 1, cdata(CDATA_HUNGER, cc) >= 12000
 		}
 	}
 	cdata(CDATA_HUNGER, cc) += nutrition
 	if ( refitem(inv(INV_ITEM_ID, ci), DBSPEC_TYPE) == FILTER_ITEM_FOOD ) {
-		if ( cc == CHARA_PLAYER ) {
-			if ( inv(INV_ITEM_PARAM2, ci) >= 3 ) {
-				healsp CHARA_PLAYER, limit(inv(INV_ITEM_LEVEL, ci) * 3 / 2 + inv(INV_ITEM_PARAM2, ci), 3, 31)
-			}
-		}
+		healsp cc, 5 + inv(INV_ITEM_PARAM2, ci) * 3 + inv(INV_ITEM_LEVEL, ci) * 3
 	}
 	if ( cdata(CDATA_ID, cc) == CREATURE_ID_CUTE_FAIRY ) {
 		if ( cdata(CDATA_RELATION, cc) == RELATION_ALLY ) {
 			if ( nutrition >= 4000 ) {
 				if ( cbit(CHARA_BIT_EXCLUSIVE_SKILL_CHARGE, cc) != FALSE ) {
diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/main.hsp ../2.10-borscht/main.hsp
--- ../2.09-borscht/main.hsp	2022-01-28 18:58:57.299880200 -0800
+++ ../2.10-borscht/main.hsp	2022-01-30 13:05:34.554519600 -0800
@@ -181,11 +181,11 @@
 	if ( dbg_randseed != 0 ) {
 		randseed = dbg_randseed
 	}
 	randomize randseed
 	exrand_randomize randseed
-	title "ElonaPlus 2.09"
+	title "ElonaPlus 2.10"
 	goto *system_init
 	goto *label_0003
 
 #deffunc netload str netload_filename
 	netrequest netload_filename
diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/map.hsp ../2.10-borscht/map.hsp
--- ../2.09-borscht/map.hsp	2022-01-28 18:58:57.381956000 -0800
+++ ../2.10-borscht/map.hsp	2022-01-30 13:05:34.677633100 -0800
@@ -1560,10 +1560,17 @@
 		map_placeplayer
 	}
 	if ( gdata(GDATA_AREA) == AREA_TRUCE_GROUND ) {
 		map_initcustom "shrine_1"
 		mdata(MDATA_MODERATE_CROWD) = 10
+		if ( gdata(GDATA_FLAG_TALKA_IN_TRUCE_GROUND) == 1 ) {
+			flt
+			characreate -1, CREATURE_ID_TALKA_THE_LONG_TWINTAIL, 16, 4
+			cdata(CDATA_ROLE, rc) = ROLE_SPECIAL
+			flt
+			itemcreate -1, ITEM_ID_AURTEHOM, 25, 16, 0
+		}
 		flt
 		itemcreate -1, ITEM_ID_ALTAR, 10, 8, 0
 		inv(INV_ITEM_PARAM1, ci) = 1
 		inv(INV_ITEM_PROPERTY, ci) = PROP_NPC
 		flt
@@ -8830,10 +8837,11 @@
 		else {
 			cxinit = -1
 			gosub *place_chara
 			cdata(CDATA_HP, rc) = cdata(CDATA_MAX_HP, rc)
 			cdata(CDATA_MP, rc) = cdata(CDATA_MAX_MP, rc)
+			cdata(CDATA_SP, rc) = cdata(CDATA_MAX_SP, rc)
 		}
 	loop
 	if ( mdata(MDATA_CAN_SAVE) == 1 ) {
 		if ( gdata(GDATA_HOUR) + gdata(GDATA_DAY) * 24 + gdata(GDATA_MONTH) * 24 * 30 + gdata(GDATA_YEAR) * 24 * 30 * 12 >= mdata(MDATA_RENEW) ) {
 			if ( mdata(MDATA_NO_RENEW) == 0 ) {
@@ -9079,10 +9087,11 @@
 				gdata(GDATA_GWEN_PITILESS_LVL) = 0
 			}
 			if ( rc >= MAX_CHARA_SAVE ) {
 				cdata(CDATA_HP, rc) = cdata(CDATA_MAX_HP, rc)
 				cdata(CDATA_MP, rc) = cdata(CDATA_MAX_MP, rc)
+				cdata(CDATA_SP, rc) = cdata(CDATA_MAX_SP, rc)
 				cdata(CDATA_SANITY, rc) = 0
 			}
 			if ( rc >= MAX_CHARA_SAVE ) {
 				if ( mdata(MDATA_RESTORE_POS) == 1 ) {
 					if ( mdata(MDATA_TILE_FILE) == 3 ) {
@@ -10111,10 +10120,19 @@
 				cdata(CDATA_EXIST, cnt) = CHAR_STATE_DEAD
 				map(cdata(CDATA_X, cnt), cdata(CDATA_Y, cnt), MAP_CHARA_INDEX_PLUS_ONE) = 0
 			loop
 		}
 	}
+	if ( adata(ADATA_ID, gdata(GDATA_AREA)) == AREA_TRUCE_GROUND ) {
+		if ( gdata(GDATA_FLAG_TALKA_IN_TRUCE_GROUND) == 0 ) {
+			if ( findchara(CREATURE_ID_TALKA_THE_LONG_TWINTAIL) != 0 ) {
+				chara_vanquish findchara(CREATURE_ID_TALKA_THE_LONG_TWINTAIL)
+			}
+			evadd EVENT_SPAWN_EVENT_NPCS
+			gdata(GDATA_FLAG_TALKA_IN_TRUCE_GROUND) = 1
+		}
+	}
 	if ( adata(ADATA_ID, gdata(GDATA_AREA)) == AREA_SISTER_MANSION ) {
 		tc = findchara(CREATURE_ID_YOUNGER_SISTER2)
 		if ( tc != CHARA_PLAYER ) {
 			if ( cdata(CDATA_PORTRAIT, tc) != 39 ) {
 				cdata(CDATA_PORTRAIT, tc) = 39
@@ -11463,43 +11481,65 @@
 						if ( limit(sdata(SKILL_NORMAL_DETECTION, cnt), 0, 100) + gdata(GDATA_TRAVEL_DISTANCE) > rnd(1000) ) {
 							cdata(CDATA_MATERIAL5_COUNTER, cnt) += 50
 						}
 					}
 				}
-				repeat 300
-					repeat
-						if ( cdata(CDATA_AREA, cnt) == gdata(GDATA_AREA) ) {
-							continue
-						}
-						if ( cdata(CDATA_EXIST, cnt) == CHAR_STATE_ALLY_WORK ) {
-							continue
-						}
-						await
-						p = rnd(400 - 100) + 100
-						if ( sdataref(SKILL_DATAREF_USE, p) != SKILL_NONE ) {
-							if ( sorg(p, tc) == 0 ) {
-								continue
-							}
-							skillexp p, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE), 1, 1000)
-							break
-						}
-					loop
+				skillexp SKILL_ATTR_STR, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				skillexp SKILL_ATTR_CON, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				skillexp SKILL_ATTR_DEX, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				skillexp SKILL_ATTR_PER, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				skillexp SKILL_ATTR_LER, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				skillexp SKILL_ATTR_WIL, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				skillexp SKILL_ATTR_MAG, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				skillexp SKILL_ATTR_CHA, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				repeat 112 - 100, 100
+					if ( sorg(cnt, tc) == 0 ) {
+						continue
+					}
+					skillexp cnt, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				loop
+				repeat 191 - 150, 150
+					if ( sorg(cnt, tc) == 0 ) {
+						continue
+					}
+					skillexp cnt, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				loop
+				repeat 302 - 300, 300
+					if ( sorg(cnt, tc) == 0 ) {
+						continue
+					}
+					skillexp cnt, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
 				loop
 			loop
 			if ( sdata(SKILL_NORMAL_TRAVELING, CHARA_PLAYER) > 0 ) {
-				skillexp SKILL_NORMAL_TRAVELING, CHARA_PLAYER, 5 + gdata(GDATA_TRAVEL_DISTANCE) * 2 / 5, 0, 1000
-				repeat 300
-					repeat
-						p = rnd(400 - 100) + 100
-						if ( sdataref(SKILL_DATAREF_USE, p) != SKILL_NONE ) {
-							if ( sorg(p, CHARA_PLAYER) == 0 ) {
-								continue
-							}
-							skillexp p, CHARA_PLAYER, limit(sdata(SKILL_NORMAL_TRAVELING, CHARA_PLAYER) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE), 1, 1000)
-							break
-						}
-					loop
+				tc = CHARA_PLAYER
+				skillexp SKILL_NORMAL_TRAVELING, tc, 5 + gdata(GDATA_TRAVEL_DISTANCE) * 2 / 5, 0, 1000
+				skillexp SKILL_ATTR_STR, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				skillexp SKILL_ATTR_CON, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				skillexp SKILL_ATTR_DEX, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				skillexp SKILL_ATTR_PER, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				skillexp SKILL_ATTR_LER, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				skillexp SKILL_ATTR_WIL, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				skillexp SKILL_ATTR_MAG, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				skillexp SKILL_ATTR_CHA, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				repeat 112 - 100, 100
+					if ( sorg(cnt, tc) == 0 ) {
+						continue
+					}
+					skillexp cnt, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				loop
+				repeat 191 - 150, 150
+					if ( sorg(cnt, tc) == 0 ) {
+						continue
+					}
+					skillexp cnt, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
+				loop
+				repeat 302 - 300, 300
+					if ( sorg(cnt, tc) == 0 ) {
+						continue
+					}
+					skillexp cnt, tc, limit(sdata(SKILL_NORMAL_TRAVELING, tc) * 1 / 4 + gdata(GDATA_TRAVEL_DISTANCE) / 2 + rnd(gdata(GDATA_TRAVEL_DISTANCE) / 2), 1, 2000)
 				loop
 			}
 			if ( p == 1 ) {
 				s = "あなた"
 			}
@@ -11524,22 +11564,35 @@
 			}
 			if ( cdata(CDATA_ALLIED, cnt) == 0 ) {
 				continue
 			}
 			tc = cnt
-			repeat 300
-				repeat
-					await
-					p = rnd(400 - 100) + 100
-					if ( sdataref(SKILL_DATAREF_USE, p) != SKILL_NONE ) {
-						if ( sorg(p, tc) == 0 ) {
-							continue
-						}
-						skillexp p, tc, limit(gdata(GDATA_DOJO_TURNS) * 4, 0, 200000)
-						break
-					}
-				loop
+			skillexp SKILL_ATTR_STR, tc, limit(gdata(GDATA_DOJO_TURNS) * 3 + rnd(gdata(GDATA_DOJO_TURNS)), 0, 200000)
+			skillexp SKILL_ATTR_CON, tc, limit(gdata(GDATA_DOJO_TURNS) * 3 + rnd(gdata(GDATA_DOJO_TURNS)), 0, 200000)
+			skillexp SKILL_ATTR_DEX, tc, limit(gdata(GDATA_DOJO_TURNS) * 3 + rnd(gdata(GDATA_DOJO_TURNS)), 0, 200000)
+			skillexp SKILL_ATTR_PER, tc, limit(gdata(GDATA_DOJO_TURNS) * 3 + rnd(gdata(GDATA_DOJO_TURNS)), 0, 200000)
+			skillexp SKILL_ATTR_LER, tc, limit(gdata(GDATA_DOJO_TURNS) * 3 + rnd(gdata(GDATA_DOJO_TURNS)), 0, 200000)
+			skillexp SKILL_ATTR_WIL, tc, limit(gdata(GDATA_DOJO_TURNS) * 3 + rnd(gdata(GDATA_DOJO_TURNS)), 0, 200000)
+			skillexp SKILL_ATTR_MAG, tc, limit(gdata(GDATA_DOJO_TURNS) * 3 + rnd(gdata(GDATA_DOJO_TURNS)), 0, 200000)
+			skillexp SKILL_ATTR_CHA, tc, limit(gdata(GDATA_DOJO_TURNS) * 3 + rnd(gdata(GDATA_DOJO_TURNS)), 0, 200000)
+			repeat 112 - 100, 100
+				if ( sorg(cnt, tc) == 0 ) {
+					continue
+				}
+				skillexp cnt, tc, limit(gdata(GDATA_DOJO_TURNS) * 3 + rnd(gdata(GDATA_DOJO_TURNS)), 0, 200000)
+			loop
+			repeat 191 - 150, 150
+				if ( sorg(cnt, tc) == 0 ) {
+					continue
+				}
+				skillexp cnt, tc, limit(gdata(GDATA_DOJO_TURNS) * 3 + rnd(gdata(GDATA_DOJO_TURNS)), 0, 200000)
+			loop
+			repeat 302 - 300, 300
+				if ( sorg(cnt, tc) == 0 ) {
+					continue
+				}
+				skillexp cnt, tc, limit(gdata(GDATA_DOJO_TURNS) * 3 + rnd(gdata(GDATA_DOJO_TURNS)), 0, 200000)
 			loop
 		loop
 		txtef COLOR_YELLOW
 		txtmore
 		txt lang("門下生は修行で少し成長しているようだ。", "miles and have gained experience.")
diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/map_rand.hsp ../2.10-borscht/map_rand.hsp
--- ../2.09-borscht/map_rand.hsp	2022-01-28 18:58:57.433003000 -0800
+++ ../2.10-borscht/map_rand.hsp	2022-01-30 13:05:34.752702400 -0800
@@ -1548,11 +1548,11 @@
 		loop
 	loop
 	return
 
 *map_riverHorizontal
-	v = 6 + gdata(GDATA_FLAG_COLLECTED_WISHES) * 2
+	v = 6 + gdata(GDATA_FLAG_COLLECTED_WISHES)
 	dy = mdata(MDATA_HEIGHT) / 2 - v / 2 - 5
 	p = 0
 	repeat mdata(MDATA_WIDTH)
 		x = cnt
 		repeat v
@@ -1594,11 +1594,11 @@
 		}
 	loop
 	return
 
 *map_riverVertical
-	w = 6 + gdata(GDATA_FLAG_COLLECTED_WISHES) * 2
+	w = 6 + gdata(GDATA_FLAG_COLLECTED_WISHES)
 	dx = mdata(MDATA_WIDTH) / 2 - w / 2 - 5
 	p = 0
 	repeat mdata(MDATA_HEIGHT)
 		y = cnt
 		repeat w
diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/proc.hsp ../2.10-borscht/proc.hsp
--- ../2.09-borscht/proc.hsp	2022-01-28 18:58:57.698248200 -0800
+++ ../2.10-borscht/proc.hsp	2022-01-30 13:05:35.127047800 -0800
@@ -5409,10 +5409,14 @@
 		if ( inv(INV_ITEM_PARAM3, ci) >= 6 ) {
 			txt lang("よくこんなの作れたな…。", "It is a surprise that was able to make it...")
 		}
 		return 1
 	}
+	if ( inv(INV_ITEM_ID, ci) == ITEM_ID_AURTEHOM ) {
+		txt lang("未実装。", "under development!")
+		return 1
+	}
 	tc = cc
 	item_identify ci, ITEM_KNOWN_NAME
 	gosub *com_book
 	return 1
 	goto *cast
@@ -5526,10 +5530,18 @@
 		gosub *effect_selectTg
 		if ( stat == 0 ) {
 			efsource = 0
 			return 0
 		}
+		if ( cdata(CDATA_SP, cc) > (-50) ) {
+			cdata(CDATA_SP, cc)--
+		}
+		if ( cc != CHARA_PLAYER ) {
+			if ( cdata(CDATA_SP, cc) > (-50) ) {
+				cdata(CDATA_SP, cc)--
+			}
+		}
 		if ( cc != CHARA_PLAYER ) {
 			if ( sdataref(SKILL_DATAREF_TYPE, efid) == 7 ) {
 				procid = (procid + 1) \ 4
 				proclist(procid) = "skSummon" + " ct:" + ct
 				if ( cdata(CDATA_RELATION, cc) == RELATION_ALLY | gdata(GDATA_AREA) == AREA_PET_ARENA ) {
@@ -5620,10 +5632,21 @@
 				}
 				efsource = 0
 				return 1
 			}
 		}
+		if ( cdata(CDATA_SP, cc) < 50 ) {
+			if ( rnd(2) == 0 ) {
+				if ( synccheck(cc, -1) ) {
+					txt lang("疲労し過ぎて失敗した！", "You are too exhausted!")
+					animeid = ANIMATION_FIZZLE
+					gosub *anime
+				}
+				efsource = 0
+				return 1
+			}
+		}
 		if ( noeffect == 1 ) {
 			txt lang("何もおきない… ", "Nothing happens...")
 			efsource = 0
 			return 1
 		}
@@ -6057,10 +6080,11 @@
 			}
 			goto *zapStaffSkip
 		}
 	}
 	efp = efp * (100 + sdata(SKILL_NORMAL_MAGIC_DEVICE, cc) * 16 + sdata(SKILL_NORMAL_MAGIC_CAPACITY, cc) * 6 + sdata(SKILL_ATTR_LER, cc) / 2) / 220
+	efpori = efp
 	hoseim = limit(cdata(CDATA_ENHANCE_SPELL, cc) * 2 / 3, 0, 1000) / 10 + limit(sdata(SKILL_ATTR_MAG, cc) / 5, 0, 300)
 	efp = efp * (100 + hoseim) / 100
 	if ( cdata(CDATA_EQUIP_WEIGHT, cc) < EQUIP_WEIGHT_CLASS_HEAVY & cdata(CDATA_EQUIP_WEIGHT, cc) >= EQUIP_WEIGHT_CLASS_MEDIUM & findbuff(cc, BUFF_MAGIC_EQUIP) != (-1) ) {
 		efp = efp * 6 / 5
 	}
diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/screen.hsp ../2.10-borscht/screen.hsp
--- ../2.09-borscht/screen.hsp	2022-01-28 18:58:57.945476500 -0800
+++ ../2.10-borscht/screen.hsp	2022-01-30 13:05:35.404303600 -0800
@@ -319,12 +319,20 @@
 			}
 			else {
 				color 0, 0, 0
 				pos sx + 14, sy
 				gcopy BUFFER_INF, 0, 440, 64, 16
-				dvpc = cdata(CDATA_DV, CHARA_PLAYER) * cdata(CDATA_DV_FIX, CHARA_PLAYER) / 100 + cdata(CDATA_DV_FIX, CHARA_PLAYER) - 100
-				pvpc = (cdata(CDATA_PV, CHARA_PLAYER) + cdata(CDATA_PV_FIX, CHARA_PLAYER) / 25) * cdata(CDATA_PV_FIX, CHARA_PLAYER) / 100
+				dvfix = cdata(CDATA_DV_FIX, CHARA_PLAYER)
+				pvfix = cdata(CDATA_PV_FIX, CHARA_PLAYER)
+				if ( cdata(CDATA_DV, CHARA_PLAYER) < 0 ) {
+					dvfix = 100
+				}
+				if ( cdata(CDATA_PV, CHARA_PLAYER) < 0 ) {
+					pvfix = 100
+				}
+				dvpc = cdata(CDATA_DV, CHARA_PLAYER) * dvfix / 100 + dvfix - 100
+				pvpc = (cdata(CDATA_PV, CHARA_PLAYER) + pvfix / 25) * pvfix / 100
 				if ( cdata(CDATA_DRAIN_HP, CHARA_PLAYER) >= 100 ) {
 					dvpc = 0
 					pvpc = 0
 				}
 				mes "" + dvpc + "/" + pvpc
@@ -8352,11 +8360,15 @@
 		}
 	}
 	if ( gdata(GDATA_TEMPORARY_FLAGS2) == 1200 ) {
 		tohit = tohit * 2
 	}
-	hoseidv = cdata(CDATA_DV, tc) * cdata(CDATA_DV_FIX, tc) / 100 + cdata(CDATA_DV_FIX, tc) - 100
+	dvfix = cdata(CDATA_DV_FIX, tc)
+	if ( cdata(CDATA_DV, tc) < 0 ) {
+		dvfix = 100
+	}
+	hoseidv = cdata(CDATA_DV, tc) * dvfix / 100 + dvfix - 100
 	if ( cdata(CDATA_DRAIN_HP, tc) >= 100 ) {
 		hoseidv = 0
 	}
 	evade = sdata(SKILL_ATTR_PER, tc) / 6 + sdata(SKILL_NORMAL_EVASION, tc) + hoseidv * 2 * 7 / 4 + 25 - cdata(CDATA_EVASION_PENALTY, tc) * 100
 	if ( cdata(CDATA_DIRECTIVE_MODE, tc) == DIRECTIVE_MODE_DEFENSIVE ) {
@@ -9013,11 +9025,15 @@
 		loop
 	}
 	if ( calcattackdmg_arg1 == 1 ) {
 		return damage
 	}
-	hoseipv = (cdata(CDATA_PV, tc) + cdata(CDATA_PV_FIX, tc) / 25) * cdata(CDATA_PV_FIX, tc) / 100
+	pvfix = cdata(CDATA_PV_FIX, tc)
+	if ( cdata(CDATA_PV, tc) < 0 ) {
+		pvfix = 100
+	}
+	hoseipv = (cdata(CDATA_PV, tc) + pvfix / 25) * pvfix / 100
 	if ( cdata(CDATA_DRAIN_HP, tc) >= 100 ) {
 		hoseipv = 0
 	}
 	prot = hoseipv + limit(sdata(carmor(tc), tc) / 4 - cdata(CDATA_PROT_PENALTY, tc) * 100, 0, 2000) + sdata(SKILL_ATTR_DEX, tc) / 10
 	if ( calcattackdmg_arg1 == 0 ) {
diff -U5 --recursive -x 'db_creature*' -x 'db_item*' '-I\*label_' ../2.09-borscht/system.hsp ../2.10-borscht/system.hsp
--- ../2.09-borscht/system.hsp	2022-01-28 18:58:57.979508100 -0800
+++ ../2.10-borscht/system.hsp	2022-01-30 13:05:35.447343300 -0800
@@ -38,11 +38,11 @@
 	}
 	gdata(GDATA_TEMPORARY_FLAGS2) = 0
 	return
 
 *fixSave
-	if ( gdata(GDATA_VERSION) != int(double("6.09") * 1000) ) {
+	if ( gdata(GDATA_VERSION) != int(double("6.10") * 1000) ) {
 		dialog lang("Ver." + gdata(GDATA_VERSION) + "のセーブデータをアップデートします。", "Updating your save data from Ver." + gdata(GDATA_VERSION) + " now.")
 	}
 	if ( gdata(GDATA_HEIR_DEED) < 0 ) {
 		gdata(GDATA_HEIR_DEED) = 1
 	}
@@ -807,15 +807,21 @@
 	if ( gdata(GDATA_VERSION) < 6090 ) {
 		if ( trait(TRAIT_NORMAL_STAT_SPD) != 0 ) {
 			cdata(CDATA_MATERIAL2_COUNTER, CHARA_PLAYER) = 1
 		}
 	}
-	if ( gdata(GDATA_VERSION) != int(double("6.09") * 1000) ) {
+	if ( gdata(GDATA_VERSION) < 6100 ) {
+		gdata(GDATA_FLAG_ABYSS_POWER) = gdata(GDATA_FLAG_ABYSS_POWER) * 20
+		if ( gdata(GDATA_FLAG_ABYSS_POWER) > 100000 ) {
+			gdata(GDATA_FLAG_ABYSS_POWER) = 100000
+		}
+	}
+	if ( gdata(GDATA_VERSION) != int(double("6.10") * 1000) ) {
 		gdata(GDATA_WORLD_RENEW) = TRUE
 	}
 	del_chara 56
-	if ( gdata(GDATA_VERSION) > int(double("6.09") * 1000) ) {
+	if ( gdata(GDATA_VERSION) > int(double("6.10") * 1000) ) {
 		dialog "invalid version"
 		goto *exit_game
 	}
 	if ( gdata(GDATA_PLAY_TIME) < 0 ) {
 		gdata(GDATA_PLAY_TIME) = 0
@@ -865,11 +871,11 @@
 		if ( gdata(GDATA_GAMEMODE_PENALTY) == 1 ) {
 			gdata(GDATA_GAMEMODE_PENALTY) = 0
 			snd SOUNDLIST_ENC
 		}
 	}
-	gdata(GDATA_VERSION) = int(double("6.09") * 1000)
+	gdata(GDATA_VERSION) = int(double("6.10") * 1000)
 	cbitmod CHARA_BIT_PCC, CHARA_PLAYER, TRUE
 	gdata(GDATA_SPEEDSUM_RESET) = 1
 	gosub *initial_recipe
 	return
 
@@ -3332,11 +3338,11 @@
 		pos x, y
 		mesbox inputlog, 600, 0, 5, val(2) * (1 + en)
 		pos x + 4, y + 4
 		gfini dx - 1, 35
 		gfdec 60, 60, 60
-		aplsel "ElonaPlus 2.09"
+		aplsel "ElonaPlus 2.10"
 		if ( stat == 1 ) {
 			dialog "Failed to get WINDOW ID", 1
 			clrobj 1
 			cfg_msg_box = 0
 			goto *prompt_word
@@ -3695,11 +3701,11 @@
 
 *game_debug
 	notesel dbm
 	buff = ""
 	if ( dbm == "" ) {
-		noteadd "ElonaPlus 2.09" + " v" + "6.09" + " Debug Console    Type \"?\" for help. Hit ESC to exit."
+		noteadd "ElonaPlus 2.10" + " v" + "6.10" + " Debug Console    Type \"?\" for help. Hit ESC to exit."
 		noteadd ""
 	}
 	font lang(cfg_font1, cfg_font2), 14 - en * 2, 0
 	objmode 2
 	pos 0, 24
@@ -4106,11 +4112,11 @@
 	objprm 1, dbm
 	objprm 2, ""
 	return
 
 *mutexobj
-	CreateMutexA 0, 0, "ElonaPlus 2.09"
+	CreateMutexA 0, 0, "ElonaPlus 2.10"
 	mutex_handle = stat
 	func_3
 	if ( stat == 183 ) {
 		dialog lang("二重起動のため終了します。", "The program is already running."), 1
 		end
