--- ./2.27.hsp	2025-06-02 04:42:20.781802500 +0800
+++ ./2.27R.hsp	2025-06-02 04:42:20.810712400 +0800
@@ -6697,11 +6697,11 @@
 #define global	SOUNDLIST_CURSE3	117
 #define global	SOUNDLIST_ENC	118
 #define global	SOUNDLIST_ENC2	119
 #define global	SOUNDLIST_PRAY2	120
 #define global	SOUNDLIST_OFFER2	121
-#define global	SOUNDLIST_MAX	SOUNDLIST_CHANDLER23 + 1
+#define global	SOUNDLIST_MAX	SOUNDLIST_DOG02 + 1
 #define global	SOUNDLIST_GONG	122
 #define global	SOUNDLIST_SLAP	123
 #define global	SOUNDLIST_CEFFECT	124
 #define global	SOUNDLIST_RAIN	125
 #define global	SOUNDLIST_WAVE	126
@@ -6885,10 +6885,12 @@
 #define global	SOUNDLIST_UMINADO	304
 #define global	SOUNDLIST_MIZU2	305
 #define global	SOUNDLIST_BURST00	306
 #define global	SOUNDLIST_GORO05	307
 #define global	SOUNDLIST_MECHA15	308
+#define global	SOUNDLIST_CAT03	309
+#define global	SOUNDLIST_DOG02	310
 
 // Group: sprite_sheet
 #define global	SPRITE_SHEET_COLS_MAP	33
 #define global	SPRITE_SHEET_ROWS_MAP	25
 #define global	SPRITE_SHEET_COLS_ITEM	33
