--- ./2.18.hsp	2023-06-10 20:14:33.531215200 +0800
+++ ./2.18R.hsp	2023-06-10 20:14:33.581083700 +0800
@@ -219,11 +219,10 @@
 #define global	AREA_VOID	42
 #define global	AREA_NT_SOUTH_BORDER	43
 #define global	AREA_SOUTH_TYRIS	44
 #define global	AREA_ST_NORTH_BORDER	45
 #define global	AREA_SMOKE_AND_PIPE	46
-#define global	AREA_TEST_WORLD	47
 #define global	AREA_TW_NORTH_BORDER	48
 #define global	AREA_MUSEUM	101
 #define global	AREA_SHOP	102
 #define global	AREA_CROP	103
 #define global	AREA_STORAGE_HOUSE	104
@@ -274,10 +273,11 @@
 #define global	AREA_DEVIL_CAPE	93
 #define global	AREA_SEBI	94
 #define global	AREA_DISCARDED_RANCH	100
 #define global	AREA_SOUTH_RASKILIS	95
 #define global	AREA_WEST_RASKILIS	96
+#define global	AREA_OCEAN	47
 
 // Group: atx
 #define global	ATX_RANDOM	0
 #define global	ATX_DUNGEON	1
 #define global	ATX_FOREST	2
@@ -346,11 +346,11 @@
 #define global	BUFF_GROW_LEARNING	24
 #define global	BUFF_GROW_WILL	25
 #define global	BUFF_GROW_MAGIC	26
 #define global	BUFF_GROW_CHARISMA	27
 #define global	BUFF_GROW_SPEED	28
-#define global	MAX_BUFF	BUFF_LIFE_TASTING + 1
+#define global	MAX_BUFF	BUFF_FLAME_OF_LIFE + 1
 #define global	BUFF_CLEAR_MIND	29
 #define global	BUFF_SUPER_REGEN	30
 #define global	BUFF_FEATHER	31
 #define global	BUFF_CHARGE	32
 #define global	BUFF_EMERGENCY_EVASION	33
@@ -386,10 +386,12 @@
 #define global	BUFF_DATA_SCAN	63
 #define global	BUFF_JYUSOU_GOUSHIN	64
 #define global	BUFF_HARMONIC_ASSAULT	65
 #define global	BUFF_LIFE_TASTING	67
 #define global	BUFF_MOGRANIC_AURA	66
+#define global	BUFF_CHAIN_OF_MANA	68
+#define global	BUFF_FLAME_OF_LIFE	69
 
 // Group: buff_type
 #define global	BUFF_TYPE_BUFF	1
 #define global	BUFF_TYPE_HEX	2
 #define global	BUFF_TYPE_FOOD	3
@@ -6086,11 +6088,11 @@
 #define global	SKILL_SPACT_CHEER	656
 #define global	SKILL_SPACT_MEWMEWMEW	657
 #define global	SKILL_SPACT_DECAPITATION	658
 #define global	SKILL_SPACT_DROP_MINE	659
 #define global	SKILL_SPACT_MANIS_DISASSEMBLY	660
-#define global	MAX_SKILL	SKILL_SPACT_SHADOW_HOP + 1
+#define global	MAX_SKILL	SKILL_SPACT_ACCEL_NECRO + 1
 #define global	SKILL_SPACT_INSANE_WAVE	661
 #define global	SKILL_SPACT_SUPER_VIBRATING_FIST	662
 #define global	SKILL_SPACT_TEMPEST_ARROW	663
 #define global	SKILL_SPACT_EXCESSIVE_HEAL	664
 #define global	SKILL_SPACT_SHINE_SNAIL	665
@@ -6280,10 +6282,11 @@
 #define global	SKILL_SPACT_HYPER_PLUG_BURST	849
 #define global	SKILL_SPACT_LIMIT_POWER	850
 #define global	SKILL_SPACT_LIMITER_VANISH	851
 #define global	SKILL_SPACT_SHADOW_JUMP	852
 #define global	SKILL_SPACT_SHADOW_HOP	853
+#define global	SKILL_SPACT_ACCEL_NECRO	854
 
 // Group: skill_spell
 #define global	STARTING_SKILL_SPELL	400
 #define global	SKILL_SPELL_HEAL_LIGHT	400
 #define global	SKILL_SPELL_HEAL_CRITICAL	401
